### PR TITLE
CR-1113671 Fix display of CMC report when not supported

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

#### Problem solved by the commit
CMC report in xbutil examine shall prints all required information only when the feature is supported by device.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1113671 CMC report prints all information even when the device not supporting this feature.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Added changes in CMC status report in XRT.

#### Risks (if any) associated the changes in the commit
Only CMC report output is changed.
 
#### What has been tested and how, request additional testing if necessary
bash-4.2# xbutil examine -r cmc -d
```
-------------------------------------------------------
1/1 [0000:d8:00.1] : xilinx_vck5000_gen4x8_xdma_base_1
-------------------------------------------------------
CMC
  Status : 0x0 (GOOD)
  Runtime clock scaling :
    Information unavailable
```

#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
bash-4.2# xbutil examine -r cmc -d

-------------------------------------------------------
1/1 [0000:d8:00.1] : xilinx_vck5000_gen4x8_xdma_base_1
-------------------------------------------------------
CMC
  Status : 0x0 (GOOD)
  Information unavailable or not applicable

**Review by rozumx — CHANGES_REQUESTED** [review]
The PR description questions are not answered. 

When they are answered, I will then review the changes.

**Comment by chvamshi-xilinx** [comment]
Hi @rajkumar-xilinx , 
Please reopen the PR after adding the description and answering the questions in the description

**Review by AShivangi — CHANGES_REQUESTED** [review]

**Review by AShivangi — APPROVED** [review]

**Review by rozumx — APPROVED** [review]
Code changes look good.

With regards to the output message, I have added SW Apps (Uday) to review it.

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp [hunk 1]
```diff
@@ -98,8 +98,15 @@ ReportCmcStatus::writeReport( const xrt_core::device* /*_pDevice*/,
   try {
     boost::property_tree::ptree cmc_scale = cmc.get_child("scaling");
     _output << boost::format("  %-22s :\n") % "Runtime clock scaling feature";
-    _output << boost::format("    %s : %s\n") % "Supported" % cmc_scale.get<std::string>("supported");
-    _output << boost::format("    %s : %s\n") % "Enabled" % cmc_scale.get<std::string>("enabled");
+    if (!cmc_scale.get<bool>("supported")) {
+      _output << "    Information unavailable or not applicable" << std::endl;
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/common/ReportCmcStatus.cpp:102]:
```suggestion
      _output << "    Information unavailable\n";

```
To match the other messages. Most other reports just state "Information Unavailable"

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp [hunk 2]
```diff
@@ -98,8 +98,15 @@ ReportCmcStatus::writeReport( const xrt_core::device* /*_pDevice*/,
...
     boost::property_tree::ptree cmc_scale = cmc.get_child("scaling");
     _output << boost::format("  %-22s :\n") % "Runtime clock scaling feature";
-    _output << boost::format("    %s : %s\n") % "Supported" % cmc_scale.get<std::string>("supported");
-    _output << boost::format("    %s : %s\n") % "Enabled" % cmc_scale.get<std::string>("enabled");
+    if (!cmc_scale.get<bool>("supported")) {
+      _output << "    Information unavailable or not applicable" << std::endl;
+      return;
+    }
+    if (!cmc_scale.get<bool>("enabled")) {
+      _output << "    Not enabled" << std::endl;
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/common/ReportCmcStatus.cpp:106]:
```suggestion
      _output << "    Not enabled\n";

```

**Review by dbenusov-xilinx — CHANGES_REQUESTED** [review]
Looks good! Just a couple things to fix

File: src/runtime_src/core/tools/common/ReportCmcStatus.cpp (hunk 2 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/tools/common/ReportCmcStatus.cpp:106]: Hi @dbenusov-xilinx , please let me know the reason to replace std::endl with "\n" ?

Comment (rajkumar-xilinx) [reply src/runtime_src/core/tools/common/ReportCmcStatus.cpp:106]: Hi @AShivangi , please let me know the right way to adding new line character. Thanks,

Comment (dbenusov-xilinx) [reply src/runtime_src/core/tools/common/ReportCmcStatus.cpp:106]: I believe our standard practice is not to use std::endl, I heard this from Soren on a previous review.

**Review by uday610 — APPROVED** [review]

**Comment by rajkumar-xilinx** [comment]
[root@localhost build]# xbutil examine -r cmc -d

```
------------------------------------------------------
1/1 [0000:04:00.1] : xilinx_u50_gen3x16_xdma_201920_3
------------------------------------------------------
CMC
  Status : 0x0 (GOOD)
  Runtime Clock Scaling :
    Information unavailable

```

**Comment by chvamshi-xilinx** [comment]
retest this please

**Comment by chvamshi-xilinx** [comment]
retest this please

**Comment by dbenusov-xilinx** [comment]
src/runtime_src/core/tools/common/ReportCmcStatus.cpp::135 has the catch statement I mentioned above

**Review by dbenusov-xilinx — APPROVED** [review]

**Comment by dbenusov-xilinx** [comment]
Thanks for the changes 👍 good job

**Pull request merged**